controllers/error.controller.js

```diff
@@ -0,0 +1,13 @@
+const handle400errors = (err, request, response, next) => {
+    if (err.status && err.msg) {
```

Comment:
This first if covers your custom errors whereas the other conditions in here cover errors from PSQL. Please separate your custom errors and PSQL errors into two handler functions as they are two different types of errors: one is errors we are triggering by rejecting Promises, the others are genuine server side errors that we must handle.